### PR TITLE
ruff: Add more rules based on Zulip's config

### PR DESCRIPTION
**Pull request by rht**
@wang-boyu this is the last one to be added to the list in https://github.com/projectmesa/mesa-geo/issues/131.

### CODE REVIEW

**Comment by wang-boyu** [comment]
> @wang-boyu this is the last one to be added to the list in [projectmesa/mesa-geo#131](https://github.com/projectmesa/mesa-geo/issues/131).

Got it. Thanks!

**Comment by rht** [comment]
I updated the earliest target of Ruff to be 3.8, because I enabled the `pyupgrade` linter.

**Comment by rht** [comment]
There was a merge conflict due to the pre-commit CI PR. But I have fixed it. @tpike3 this PR is ready for review. I'm not adding anything more.

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,29 @@
 [tool.ruff]
-select = ["E", "F", "I"]
+select = [
```
Comment (tpike3) [inline pyproject.toml:2]: similar to [Zulip](https://github.com/zulip/zulip/blob/201debc1ea28a489b1c235d4936405b11d78d4b4/pyproject.toml#L100) and the extend-ignore comments below (line 27) can you add the reference to the annotations. 

Also for purely out of curiosity why did you use Zulip as the model and not Pandas or FastAPI?

Comment (rht) [reply pyproject.toml:3]: Added. 2 reasons: 1. because Zulip's config is more comprehensive, 2. I heard about Ruff from Zulip devs.

Comment (tpike3) [reply pyproject.toml:3]: Ver cool, thanks!

**Pull request merged**